versioned/spi/src/main/java/org/projectnessie/versioned/store/ContentSerializerBundle.java

```diff
@@ -15,9 +15,12 @@
  */
 package org.projectnessie.versioned.store;
 
-import java.util.function.Consumer;
-
-/** Implementations of this {@link java.util.ServiceLoader service} register content serializers. */
+/**
+ * Custom content serializers need to implement this interface and call {@link
+ * ContentSerializerRegistry#register(ContentSerializer)} for each serializer.
+ *
+ * <p>Instances of this class are loaded via Java's {@link java.util.ServiceLoader}.
+ */
 public interface ContentSerializerBundle {
-  void register(Consumer<ContentSerializer<?>> registry);
+  void register(ContentSerializerRegistry registry);
```

Comment:
Well, unlike `Content.Type`, here we register serializer instances... Could you simply do `ServiceLoader.load(ContentSerializer.class)`? On the receiving side, we do not know specific type parameters anyway.

Reply:
But why have another approach here?

Reply:
Do remove an extra interface... but this is a minor point, optional.